### ISSUE TITLE
[Bug]: NVFP4 inference appears to cause GPU memory leaks, 5090 inference llama 2 13b OOM

### ISSUE THREAD

**Issue by alvinshao0313** [vllm-project/llm-compressor#1769]
### ⚙️ Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
### Environment Information ###
Operating System: `Linux-5.15.0-94-generic-x86_64-with-glibc2.35`
Python Version: `3.11.13 (main, Jun  5 2025, 13:12:00) [GCC 11.2.0]`
llm-compressor Version: `0.7.1.dev1+gba72530a`
compressed-tensors Version: `0.11.1a20250820`
transformers Version: `4.53.0`
torch Version: `2.8.0`
CUDA Devices: `['NVIDIA GeForce RTX 5090']`
AMD Devices: `None`
```

</details>

### 🐛 Describe the bug

When I tested the llama 2 13b of NVFP4 using the code with a 5090, OOM occurred.

```python

@torch.no_grad()
def calculate_wikitext2_ppl_standard(model, model_path, seed=0, limit=None):
    try:
        seqlen = 2048

        testloader = get_wikitext2_test(seed=seed, seqlen=seqlen, model=model_path)
        testenc = testloader.input_ids

        nsamples = testenc.numel() // seqlen

        if limit is not None:
            nsamples = min(nsamples, limit)

        use_cache = model.config.use_cache
        model.config.use_cache = False
        model.eval()

        nlls = []

        with torch.no_grad():
            pbar = tqdm(range(nsamples))
            for i in pbar:
                batch = testenc[:, (i * seqlen): ((i + 1) * seqlen)].to('cuda')

                outputs = model.model(batch)
                hidden_states = outputs[0]
                logits = model.lm_head(hidden_states)

                shift_logits = logits[:, :-1, :]
                shift_labels = testenc[:, (i * seqlen): ((i + 1) * seqlen)][:, 1:].to(model.lm_head.weight.device)

                loss_fct = nn.CrossEntropyLoss()
                loss = loss_fct(
                    shift_logits.view(-1, shift_logits.size(-1)),
                    shift_labels.view(-1),
                )

                neg_log_likelihood = loss.float()

                if not math.isnan(neg_log_likelihood):
                    nlls.append(neg_log_likelihood)

                if nlls:
                    ppl = torch.exp(torch.stack(nlls).mean())
                    pbar.set_description(f"PPL: {ppl.item():.2f}")

                if limit is not None and i >= limit:
                    break

        if nlls:
            ppl = torch.exp(torch.stack(nlls).mean())
            final_ppl = ppl.item()
        else:
            return None

        model.config.use_cache = use_cache

        logging.info(f'wikitext ppl : {final_ppl:.2f}')

        results = {
            'wiki_ppl': final_ppl,
            'valid_samples': len(nlls),
            'total_samples': nsamples,
            'seqlen': seqlen
        }

        return results
```
However, using meta-llama/Llama-2-13b-hf does not, which is unreasonable, because the basic requirement of NVFP4 is to reduce GPU memory.

Additionally, it was observed on llama 2 7b nvfp4 that there is a significant fluctuation in GPU memory usage under FP16 mode.  
Through my in-depth research, I found that there is a noticeable increase in GPU memory usage after each softmax, o_proj, and mlp computation, which should be related to the compute kernel issue.

By the way, the quantitative code follows what is written in the example. Small models can be 推理 using a 5090, but the GPU memory fluctuation is larger than FP16

**Comment by brian-dellabetta** [comment]
Hi @alvinshao0313 , just to confirm, are you loading the model up in huggingface or vllm? You won't see any memory decrease in huggingface, in fact it will be slightly worse because of the additional scales

**Comment by alvinshao0313** [comment]
I am using a model loaded with transformers, not using VLLM for inference

**Comment by dsikka** [comment]
Hi @alvinshao0313 - loading the nvfp4 model through transformers will not provide any performance benefit. The forward passes are computed using the dense model, as there is no support for kernels to support or optimize quantized inference for compressed-tensors. The pathway through transformers has primarily been created to test model accuracy, not enhance model performance. 

For that, you'll need to use an inference engine that has compressed-tensors support, such as vLLM. Running the model through vLLM will leverage the compression benefits for optimized inference.

**Comment by zeralyngkhoi19** [comment]
@dsikka @brian-dellabetta just to confirm Is this the same for all scheme like W8A8 or W4A16 and methods like GPTQ and AWQ? Since I saw same  and sometimes greater GPU memory usage in models quantized using llm-compressor  as compared to  base model when loading them using huggingface transformers
